packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ConvenienceMethodTemplateBase.java

```diff
@@ -831,9 +832,20 @@ private static boolean isMultipartModel(IType type) {
             String name = convenienceParameter.getSerializedName();
             parameterMap.put(convenienceParameter, clientParameters.get(name));
         }
+        if (isPageStreamingClientMethodType(convenienceMethod.getType())) {
+            final MethodPageDetails pageDetails = convenienceMethod.getMethodPageDetails();
+            if (pageDetails != null) {
+                parameterMap.entrySet()
+                    .removeIf(it -> pageDetails.shouldHideParameter(it.getKey().getClientMethodParameter()));
+            }
+        }
         return parameterMap;
     }
 
+    private static boolean isPageStreamingClientMethodType(ClientMethodType type) {
```

Comment:
move`isPageStreamingClientMethodType` to util, same function duplicated in 3 places.

Reply:
moved to `ClientMethod`